frontend/src/admin/user-form.tsx

```diff
@@ -0,0 +1,142 @@
+    };
+
+    const handleInputChange =(e: React.ChangeEvent<HTMLInputElement>) => {
+        const { name, value, type, checked } = e.target;
+        setUser((prev) => ({...prev,
+            [name]: type === "checkbox" ? checked : value,
+        } as User));
+    };
+
+    useEffect(() => {
+        if (userId) {
+            const fetchUser = async () => {
+                setLoading(true);
+                try {
+                    const response = await fetch(`${import.meta.env.VITE_ZTOR_URL}/users/${userId}`, {
+                        credentials: "include",
+                    });
+                    if (response.status === 401) {
+                        redirectToLogin();
+                        return;
+                    }
+                    if (response.ok) {
+                        const userData = await response.json();
+                        setUser(userData);
+                        setOriginalData(userData);
+                    } else {
+                        alert(`Error fetching user: ${response.statusText}`);
+                    }
+                } catch (error) {
+                    alert((error as Error).message);
+                } finally {
+                    setLoading(false);
+                }
+            };
+            fetchUser();
+        } else {
+            setIsEditing(true);
+        }
+    }, [userId]);
+
+    const handleSubmit = async (event: FormEvent) => {
+        event.preventDefault();
+        setLoading(true);
+        try {
+            const method = userId ? "PUT" : "POST";
+            const url = `${import.meta.env.VITE_ZTOR_URL}/users`
+            const response = await fetch(url, {
+                method,
+                headers: {
+                    "Content-Type": "application/json",
+                },
+                credentials: "include",
+                body: JSON.stringify(user),
+            });
+            if (response.status === 401) {
+                redirectToLogin();
+                return;
+            }
+
+            if (response.ok) {
+                navigate(`/users`);
+            } else {
+                const errorData = await response.json();
+                alert(`Error: ${errorData.message}`);
+            }
+        } finally {
+            setIsEditing(false);
+            setLoading(false);
+        }
+    };
+
+    return (
+        <PrimeReactProvider>
+            <div style={{ padding: "20px", maxWidth: "500px", margin: "0 auto" }}>
+                <h3>{userId ? "Edit User" : "Add User"}</h3>
+                <form
+                    onSubmit={handleSubmit}
+                    style={{ display: "flex", flexDirection: "column", gap: "10px" }}
+                >
+                    <label htmlFor="name">Note:</label>
+                    <InputText
+                        id="note"
+                        name="note"
+                        value={user?.note || ""}
+                        onChange={handleInputChange}
```

Comment:
Why aren't you using uncontrolled components and FormData like we did in local4local? It seems simpler.

React 19 adds `useActionState` to make this even easier: https://react.dev/reference/react/useActionState. We can probably update. 

Doesn't need to be changed in this PR but give it some thought.